### PR TITLE
feat: Pagination

### PR DESCRIPTION
**Pull request by Harshavardhan11**
Merged Conflicts

### CODE REVIEW

File: src/main/java/com/example/FastDataSearch/service/DataIndexService.java [hunk 1]
```diff
@@ -98,27 +102,38 @@ public List<String> getColumns(String indexName) {
...
+                .stream()
+                .map(hit -> (Map<String, Object>) hit.getContent())
+                .collect(Collectors.toList());
+
+        long totalHits = searchHits.getTotalHits();
+        Page<Map<String, Object>> resultsPage = new PageImpl<>(results, pageable, totalHits);
+
+        System.out.println("LOG 10: Query executed successfully. Results count: " + results.size());
+
+        return Map.of(
```
Comment (flow6979) [inline src/main/java/com/example/FastDataSearch/service/DataIndexService.java:134]: put tis part in controller only

**Review by flow6979 — APPROVED** [review]

**Comment by flow6979** [comment]
@Bhargav1603 @Vaishnavi-Kahar approve pls

**Review by Vaishnavi-Kahar — APPROVED** [review]

**Comment by flow6979** [comment]
wait i am checking for list of queries

**Comment by flow6979** [comment]
i'll merge it, if that functionality works fine.

**Comment by flow6979** [comment]
it is throwing 500 error for list of queries. let me see if i can do something

**Comment by Harshavardhan11** [comment]
Added final changes

**Pull request merged**